dhis-2/dhis-services/dhis-service-program-rule/src/main/java/org/hisp/dhis/programrule/action/validation/ProgramRuleActionValidationContextLoader.java

```diff
@@ -72,6 +76,12 @@ public ProgramRuleActionValidationContext load( Preheat preheat, PreheatIdentifi
 
         Program program = preheat.get( preheatIdentifier, Program.class, rule.getProgram() );
 
+        if ( program == null )
```

Comment:
Just wondering, do we know the root cause? This seems a little bit like a patch/workaround, though I can be work. Could we figure out why the program is not part of the preheat in the first place?

Reply:
@larshelge Please check the attached payload. User wants to update ProgramRuleAction, so they only post only that object with all the necessary references. 
However with current model design, we have ProgramRule -- ProgramRuleAction, so there no link to Program.

Reply:
Okay understand. Would there be a benefit in adding the program to the bundle in the beginning of the process, to avoid fetching it many times or similar? Just checking, might be wrong.

Reply:
Yes, I'm working on new Preheat hook so developers can add on demand objects UID to the preheat process. Currently we will need to add an if-else block to the `PreheatService` which is not good.